File: source/qa/cxx/test/scalability/array_experiment_test.cpp
#define BOOST_TEST_MODULE lue qa scalability instrument array_experiment
#include "lue/qa/scalability/instrument/array_experiment.hpp"
#include "lue/stream.hpp"
#include <boost/test/included/unit_test.hpp>


BOOST_AUTO_TEST_CASE(use_case_01)
{
    lue::qa::Count const nr_workers{5};
    lue::qa::Shape const array_shape{60, 40};
    lue::qa::Shape const partition_shape{10, 10};
    lue::qa::ArrayExperiment array_experiment{nr_workers, array_shape, partition_shape};

    BOOST_CHECK_EQUAL(array_experiment.nr_workers(), nr_workers);
    BOOST_CHECK_EQUAL(array_experiment.array_shape(), array_shape);
    BOOST_CHECK_EQUAL(array_experiment.partition_shape(), partition_shape);

    array_experiment.start();

    lue::qa::Count const nr_runs{3};

    for (lue::qa::Count run_idx = 0; run_idx < nr_runs; ++run_idx)
    {
        lue::qa::Run run{};

        run.start();
        run.stop();

        array_experiment.add(run);
    }

    array_experiment.stop();

    BOOST_CHECK_EQUAL(std::size(array_experiment.runs()), nr_runs);
}

File: source/quality_assurance/python/test/lue_qa_test/scalability/instrument/array_experiment_test.py
import unittest

import lue.qa.scalability.instrument as lqi


class ArrayExperimentTest(unittest.TestCase):
    def test_use_case_01(self):

        nr_workers = 5
        array_shape = (60, 40)
        partition_shape = (10, 10)
        experiment = lqi.ArrayExperiment(nr_workers, array_shape, partition_shape)

        self.assertEqual(experiment.nr_workers, nr_workers)
        self.assertEqual(experiment.array_shape, array_shape)
        self.assertEqual(experiment.partition_shape, partition_shape)

        experiment.start()
        experiment.stop()
